rpc-client/src/nonblocking/rpc_client.rs

```diff
@@ -2179,8 +2179,19 @@ impl RpcClient {
         &self,
         commitment: CommitmentConfig,
         max_stake_percent: f32,
+    ) -> ClientResult<()> {
+        self.wait_for_max_stake_below_threshold_with_timeout(commitment, max_stake_percent, None)
```

Comment:
I'm definitely pro-code reuse tho; if we want to keep the timeout around, maybe a non-pub helper that takes an `Option<Duration>`, and then:
- `wait_for_max_stake_below_threshold` calls helper with `None`
- `wait_for_max_stake_below_threshold_with_timeout` calls helper with `Some(timeout)`